Library/Homebrew/test/support/helper/spec/integration_command.rb

```diff
@@ -0,0 +1,92 @@
+require "open3"
+
+RSpec::Matchers.define_negated_matcher :not_to_output, :output
+RSpec::Matchers.define_negated_matcher :be_a_failure, :be_a_success
+
+RSpec::Matchers.define :be_a_success do
+  match do |actual|
+    if actual.is_a?(Proc)
+      status = actual.call
+    else
+      status = actual
+    end
+
+    status.success?
+  end
+
+  def supports_block_expectations?
+    true
+  end
+
+  # It needs to be nested like this:
+  #
+  #   expect {
+  #     expect {
+  #       # command
+  #     }.to be_a_success
+  #   }.to output(something).to_stdout
+  #
+  # rather than this:
+  #
+  #   expect {
+  #     expect {
+  #       # command
+  #     }.to output(something).to_stdout
+  #   }.to be_a_success
+  #
+  def expects_call_stack_jump?
+    true
+  end
+end
+
+RSpec.configure do |config|
+  config.before(:each) do |example|
+    (HOMEBREW_PREFIX/"bin").mkpath
+    FileUtils.touch HOMEBREW_PREFIX/"bin/brew"
+    @example = example
```

Comment:
Does RSpec not expose `#full_description` in a test body? Storing the example as an instance variable seems pretty hacky.